marshal/tpms-types.c

```diff
@@ -243,6 +243,39 @@ static TSS2_RC unmarshal_tagged_pcr_selection(uint8_t const buffer[], size_t buf
     return TSS2_RC_SUCCESS;
 }
 
+#define TPMS_MARSHAL_0(type) \
+TSS2_RC Tss2_MU_##type##_Marshal(type const *src, uint8_t buffer[], \
+                                 size_t buffer_size, size_t *offset) \
+{ \
+	 if (offset == NULL) { \
+                return TSS2_SYS_RC_BAD_REFERENCE; \
+	 } \
+    if (!src) { \
```

Comment:
Above we do if (offset == NULL) and here if (!src) please keep it consistent.
Can we just do if (!offset || !src) return TSS2_SYS_RC_BAD_REFERENCE;

Reply:
Testing offset against NULL was actually by itself a mistake...
fixed